### PR TITLE
Notify slack when k8s jobs fail

### PR DESCRIPTION
**Pull request by jweibel22**
Make slack notifications when a k8s job fails

### CODE REVIEW

**Comment by jweibel22** [comment]
I'm not sure how to best test this? It looks like the e2e tests that exists are for checking the flux stuff....

File: cmd/daemon/kubernetes/jobs.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				// Notify the release-manager with the job error event.
+				err = c.exporter.SendJobErrorEvent(ctx, http.JobErrorEvent{
+					JobName:     job.Name,
+					Namespace:   job.Namespace,
+					Errors:      jobErrorMessages(job),
+					ArtifactID:  job.Annotations["lunarway.com/artifact-id"],
+					AuthorEmail: job.Annotations["lunarway.com/author"],
+				})
+				if err != nil {
+					log.Errorf("Failed to send successful job error event: %v", err)
```
Comment (Crevil) [inline cmd/daemon/kubernetes/jobs.go:52]: `successful job error`? 🙈 
```suggestion
					log.Errorf("Failed to send job error event: %v", err)
```

File: cmd/server/http/daemon_k8s_job_error_webhook.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		if err != nil {
+			logger.Errorf("http: daemon k8s job error webhook: decode request body failed: %v", err)
+			invalidBodyError(w)
+			return
+		}
+		logger = logger.WithFields("event", event)
+		err = flowSvc.NotifyK8SJobErrorEvent(ctx, &event)
+		if err != nil && errors.Cause(err) != slack.ErrUnknownEmail {
+			logger.Errorf("http: daemon k8s pod error webhook failed: %+v", err)
+		}
```
Comment (Crevil) [inline cmd/server/http/daemon_k8s_job_error_webhook.go:31]: No return here or response writing?

File: cmd/server/http/daemon_k8s_job_error_webhook.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		w.WriteHeader(http.StatusOK)
+		err = payload.encodeResponse(ctx, w, httpinternal.KubernetesNotifyResponse{})
+		if err != nil {
+			logger.Errorf("http: daemon k8s job error webhook: environment: '%s' marshal response: %v", event.Environment, err)
+		}
```
Comment (Crevil) [inline cmd/server/http/daemon_k8s_job_error_webhook.go:36]: Same as above regarding response ?

File: internal/http/types.go [hunk 4]
```diff
@@ -115,6 +115,21 @@ type PodErrorEvent struct {
 	Environment string           `json:"environment,omitempty"`
 	ArtifactID  string           `json:"artifactId,omitempty"`
 }
+
+type JobConditionError struct {
+	Reason          string `json:"reason,omitempty"`
+	ErrorMessage 	string `json:"message,omitempty"`
```
Comment (Crevil) [inline internal/http/types.go:121]: Is the difference in json key and Go field intentional? I would expect them to be alligned?

**Review by Crevil — APPROVED** [review]
Some minor things regarding error checks in the HTTP handler.

Other than that, LGTM. 🎉

File: internal/http/types.go (hunk 4 above)
Comment (jweibel22) [reply internal/http/types.go:121]: no not intentional!

**Comment by jweibel22** [comment]
I've tested this locally and got an auth error from slack but data looks ok:
```
$ make e2e-setup
$ k apply -f ~/Downloads/failingjob.yaml
$ k logs release-manager-5897948bf5-76nlb
{"level":"error","@timestamp":"2021-04-08T09:29:32.598Z","caller":"http/http.go:133","message":"http: daemon k8s pod error webhook failed: not_authed\npost k8s NotifyK8SJobErrorEvent slack message","fields":{"event":{"jobName":"pi","namespace":"default","errors":[{"reason":"BackoffLimitExceeded","message":"Job has reached the specified backoff limit"}],"authorEmail":"jwr@lunar.app","environment":"local","artifactId":"hamctl-support-2-d8f80b5ba6-d8f80b5ba6"}}
```

**Pull request merged**